src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseSearchValuesTests.cs

```diff
@@ -0,0 +1,1023 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpUseSearchValuesAnalyzer,
+    Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpUseSearchValuesFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class UseSearchValuesTests
+    {
+        [Fact]
+        public async Task TestIndexOfAnyAnalyzer()
```

Comment:
We generally discourage adding an analyzer-only test that verifies just the diagnostics but not the fixed code. Is there a reason we can't switch over to using `VerifyCodeFixAsync` for all the tests being added here? That will also avoid duplicate testing, as I see you have smaller tests later in the file that is verifying the fixed source (or lack of where there is no diagnostic or fix is not supported).

I'll let @buyaa-n sync up with you and make a call on whether we are ok with the current testing pattern of separate analyzer and code fix tests in this file OR want to switch over to `VerifyCodeFixAsync` for all tests.

Reply:
I kept the analyzer-only test mainly to also test all the cases where we choose not to report the diagnostic.
E.g. having tests like these side-by-side
```c#
_ = chars.IndexOfAny("aeiou");
_ = chars.IndexOfAny([|"aeiouA"|]);

_ = chars.IndexOfAny([|new char[] { 'a', 'e', 'i', 'o', 'u', 'A' }|]);
_ = chars.IndexOfAny(new[] { 'a', 'e', 'i', 'o', 'u', NonConstChar });
```

The code fixer tests below should be exercising most/all of the patterns in this test that we do flag.